src/compiler/program.ts

```diff
@@ -690,6 +696,14 @@ namespace ts {
         let program: Program;
         let files: SourceFile[] = [];
         let fileProcessingDiagnostics = createDiagnosticCollection();
+        const currentDirectory = host.getCurrentDirectory();
+        const resolvedLibraries: Map<ResolvedLibrary> = {};
+        let libraryRoot =
+            (options.rootDir && ts.toPath(options.rootDir, currentDirectory, host.getCanonicalFileName)) ||
+            (options.configFilePath && getDirectoryPath(getNormalizedAbsolutePath(options.configFilePath, currentDirectory)));
+        if (libraryRoot === undefined) {
+            libraryRoot = computeCommonSourceDirectoryOfFilenames(rootNames);

```

Comment:
this is kinda unexpected. so `tsc a.ts` will result in a different search location from `tsc a\a.ts b\b.ts`, even if `a.ts` has a reference to `b.ts`?
I would say make it an error to try to resolve a library reference, without a root, and ask the user to specify `rootDir`.